packages/payload/src/admin/components/views/collections/List/Cell/field-types/Relationship/index.tsx

```diff
@@ -68,11 +69,18 @@ const RelationshipCell: React.FC<CellComponentProps<RelationshipField>> = (props
           i18n,
         })
 
+        let fileField = null
+        if ((field as unknown as UploadField).type === 'upload' && document) {
```

Comment:
If you change the component type you wont need this awkward check.

```tsx
const RelationshipCell: React.FC<CellComponentProps<RelationshipField | UploadField>>
```

And then you can just use

```ts
if (field.type === 'upload' && document)
```

Reply:
fixed